sentry/src/main/java/io/sentry/transport/BlockingHttpTransport.java

```diff
@@ -11,57 +11,79 @@
-      final int maxQueueSize,
-      final SentryOptions options,
-      final @NotNull RateLimiter rateLimiter) {
+  public BlockingHttpTransport(
+      final @NotNull SentryOptions options,
+      final @NotNull RateLimiter rateLimiter,
+      final @NotNull ITransportGate transportGate,
+      final @NotNull IConnectionConfigurator connectionConfigurator,
+      final @NotNull URL sentryUrl) {
     this(
-        transport,
-        transportGate,
-        envelopeCache,
-        initExecutor(maxQueueSize, envelopeCache, options.getLogger()),
+        initExecutor(
+            options.getMaxQueueSize(), options.getEnvelopeDiskCache(), options.getLogger()),
         options,
-        rateLimiter);
+        rateLimiter,
+        transportGate,
+        new HttpConnection(options, connectionConfigurator, sentryUrl, rateLimiter));
   }
 
-  @TestOnly
-  AsyncConnection(
-      final @NotNull ITransport transport,
-      final @NotNull ITransportGate transportGate,
-      final @NotNull IEnvelopeCache envelopeCache,
-      final @NotNull ExecutorService executorService,
+  public BlockingHttpTransport(
+      final @NotNull ExecutorService executor,
       final @NotNull SentryOptions options,
-      final @NotNull RateLimiter rateLimiter) {
-    this.transport = transport;
-    this.transportGate = transportGate;
-    this.envelopeCache = envelopeCache;
-    this.options = options;
-    this.executor = executorService;
-    this.rateLimiter = rateLimiter;
+      final @NotNull RateLimiter rateLimiter,
+      final @NotNull ITransportGate transportGate,
+      final @NotNull HttpConnection httpConnection) {
+    this.executor = Objects.requireNonNull(executor, "executor is required");
+    this.envelopeCache =
+        Objects.requireNonNull(options.getEnvelopeDiskCache(), "envelopeCache is required");
+    this.options = Objects.requireNonNull(options, "options is required");
+    this.rateLimiter = Objects.requireNonNull(rateLimiter, "rateLimiter is required");
+    this.transportGate = Objects.requireNonNull(transportGate, "transportGate is required");
+    this.connection = Objects.requireNonNull(httpConnection, "httpConnection is required");
+  }
+
+  @Override
+  @SuppressWarnings("FutureReturnValueIgnored")
+  public void send(SentryEnvelope envelope, Object hint) throws IOException {
+    // For now no caching on envelopes
+    IEnvelopeCache currentEnvelopeCache = envelopeCache;
+    boolean cached = false;
+    if (hint instanceof Cached) {
+      currentEnvelopeCache = NoOpEnvelopeCache.getInstance();
+      cached = true;
+      options.getLogger().log(SentryLevel.DEBUG, "Captured Envelope is already cached");
+    }
+
+    final SentryEnvelope filteredEnvelope = rateLimiter.filter(envelope, hint);
+
+    if (filteredEnvelope == null) {
+      if (cached) {
+        envelopeCache.discard(envelope);
+      }
+    } else {
+      executor.submit(new EnvelopeSender(filteredEnvelope, hint, currentEnvelopeCache));
```

Comment:
I think the name `BlockingHttpTransport` is misleading since it gives the impression the interface is going to block until for the event submission, which isnt' the case.

We're scheduling the event through the executor here.

Maybe we can use another name?

Reply:
yeah I kinda agree, not sure about a suitable name though.

SingleThreadHttpTransport?

Reply:
Can be also `AsyncHttpTransport`. It is async, but blocking.